src/mito2/src/sst.rs

```diff
@@ -241,3 +246,92 @@ fn plain_internal_fields() -> [FieldRef; 2] {
         Arc::new(Field::new(OP_TYPE_COLUMN_NAME, ArrowDataType::UInt8, false)),
     ]
 }
+
+/// Gets the estimated number of series from record batches.
+///
+/// This struct tracks the last timestamp value to detect series boundaries
+/// by observing when timestamps decrease (indicating a new series).
+#[derive(Default)]
+pub(crate) struct SeriesEstimator {
+    /// The last timestamp value seen
+    last_timestamp: Option<i64>,
+    /// The estimated number of series
+    series_count: u64,
+}
+
+impl SeriesEstimator {
+    /// Updates the estimator with a new Batch.
+    ///
+    /// Since each Batch contains only one series, this increments the series count
+    /// and updates the last timestamp.
+    pub(crate) fn update(&mut self, batch: &Batch) {
+        let Some(last_ts) = batch.last_timestamp() else {
+            return;
+        };
+
+        // Checks if there's a boundary between the last batch and this batch
+        if let Some(prev_last_ts) = self.last_timestamp {
+            // If the first timestamp of this batch is less than the last timestamp
+            // we've seen, it indicates a new series
+            if let Some(first_ts) = batch.first_timestamp()
+                && first_ts.value() <= prev_last_ts
+            {
+                self.series_count += 1;
+            }
+        } else {
+            // First batch, counts as first series
+            self.series_count = 1;
+        }
+
+        // Updates the last timestamp
+        self.last_timestamp = Some(last_ts.value());
+    }
+
+    /// Updates the estimator with a new record batch in flat format.
+    ///
+    /// This method examines the time index column to detect series boundaries.
+    pub(crate) fn update_flat(&mut self, record_batch: &RecordBatch) {
+        let batch_rows = record_batch.num_rows();
+        if batch_rows == 0 {
+            return;
+        }
+
+        let time_index_pos = time_index_column_index(record_batch.num_columns());
+        let timestamps = record_batch.column(time_index_pos);
+        let Some((ts_values, _unit)) = timestamp_array_to_primitive(timestamps) else {
+            return;
+        };
+        let values = ts_values.values();
+
+        // Checks if there's a boundary between the last batch and this batch
+        if let Some(last_ts) = self.last_timestamp {
+            if values[0] < last_ts {
```

Comment:
Not `<=`? I noticed that line 277 is `<=`.

Reply:
You're right.